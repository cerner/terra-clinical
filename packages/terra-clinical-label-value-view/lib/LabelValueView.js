'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

require('terra-base/lib/baseStyles');

require('./LabelValueView.scss');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var propTypes = {
  /**
   * The label of the LabelValueView.
   */
  label: _propTypes2.default.string.isRequired,
  /**
  * The text to be displayed underneath the label with the provided styling.
  */
  textValue: _propTypes2.default.string,
  /**
   *  Child component(s) to display underneath the label.
   */
  children: _propTypes2.default.node
};

var defaultProps = {
  textValue: '',
  children: undefined
};

var LabelValueView = function LabelValueView(_ref) {
  var label = _ref.label,
      textValue = _ref.textValue,
      children = _ref.children,
      customProps = _objectWithoutProperties(_ref, ['label', 'textValue', 'children']);

  var labelValueViewClassNames = (0, _classnames2.default)('terraClinical-LabelValueView', _defineProperty({}, '' + customProps.className, customProps.className));

  var textValueSection = void 0;
  if (!textValue && !children) {
    textValueSection = _react2.default.createElement(
      'div',
      { className: 'terraClinical-LabelValueView-value' },
      '--'
    );
  } else if (textValue) {
    textValueSection = _react2.default.createElement(
      'div',
      { className: 'terraClinical-LabelValueView-value' },
      textValue
    );
  }

  return _react2.default.createElement(
    'div',
    _extends({}, customProps, { className: labelValueViewClassNames }),
    _react2.default.createElement(
      'div',
      { className: 'terraClinical-LabelValueView-label' },
      label
    ),
    textValueSection,
    children
  );
};

LabelValueView.propTypes = propTypes;
LabelValueView.defaultProps = defaultProps;

exports.default = LabelValueView;