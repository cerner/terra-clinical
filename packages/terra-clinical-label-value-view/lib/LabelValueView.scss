@import '~terra-mixins';

.terraClinical-LabelValueView {
  border: 0;
  padding: 0.1785rem;
}

.terraClinical-LabelValueView-label {
  color: $terra-grey-50;
  font-size: 1rem;
  line-height: 1.428;
}

.terraClinical-LabelValueView-value {
  font-size: 1.428rem;
  line-height: 1.428;
}
