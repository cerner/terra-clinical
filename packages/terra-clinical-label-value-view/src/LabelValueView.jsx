import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import styles from './LabelValueView.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * The label of the LabelValueView.
   */
  label: PropTypes.string.isRequired,
  /**
  * The text to be displayed underneath the label with the provided styling.
  */
  textValue: PropTypes.string,
  /**
   *  Child component(s) to display underneath the label.
   */
  children: PropTypes.node,
};

const defaultProps = {
  textValue: '',
  children: undefined,
};

const LabelValueView = ({
  label, textValue, children, ...customProps
}) => {
  let textValueSection;
  if (!textValue && !children) {
    textValueSection = <div className={cx('value')}>--</div>;
  } else if (textValue) {
    textValueSection = <div className={cx('value')}>{textValue}</div>;
  }

  return (
    <div {...customProps} className={cx('label-value-view', customProps.className)}>
      <div className={cx('label')}>{label}</div>
      {textValueSection}
      {children}
    </div>
  );
};

LabelValueView.propTypes = propTypes;
LabelValueView.defaultProps = defaultProps;

export default LabelValueView;
