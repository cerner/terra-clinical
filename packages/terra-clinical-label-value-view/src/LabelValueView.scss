/* stylelint-disable selector-class-pattern  */
:local {
  .label-value-view {
    border: 0;
    padding: 0.1785rem;
  }

  .label {
    color: #909496;
    font-size: 1rem;
    line-height: 1.428;
  }

  .value {
    font-size: 1.428rem;
    line-height: 1.428;
  }
}
