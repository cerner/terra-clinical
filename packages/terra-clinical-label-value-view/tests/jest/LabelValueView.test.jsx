import React from 'react';
import LabelValueView from '../../src/LabelValueView';

describe('LabelValueView', () => {
  const defaultView = <LabelValueView label="Label" />;

  // Snapshot Tests
  it('should render a default LabelValueView', () => {
    const wrapper = shallow(defaultView);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a LabelValueView for text input', () => {
    const textInputView = (
      <LabelValueView label="Label" textValue="Sample Text" />
    );
    const wrapper = shallow(textInputView);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render LabelValueView for an node input', () => {
    const nodeInputView = (
      <LabelValueView label="Label">
        <div>More Sample Text</div>
      </LabelValueView>
    );
    const wrapper = shallow(nodeInputView);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render LabelValueView for text and node inputs', () => {
    const multipleInputsView = (
      <LabelValueView label="Label" textValue="Sample Text 1">
        <div>More Sample Text</div>
      </LabelValueView>
    );
    const wrapper = shallow(multipleInputsView);
    expect(wrapper).toMatchSnapshot();
  });

  // Prop Tests
  it('should set the label text', () => {
    const wrapper = shallow(defaultView);
    expect(wrapper.find('.label').text()).toEqual('Label');
  });

  it('should use the default value when no value is given', () => {
    const wrapper = shallow(defaultView);
    expect(wrapper.find('.value').text()).toEqual('--');
  });

  // Structure Tests
  it('should have the class terraClinical-LabelValueView', () => {
    const wrapper = shallow(defaultView);
    expect(wrapper.prop('className')).toContain('label-value-view');
  });

  it('should have the class terraClinical-LabelValueView-label', () => {
    const wrapper = shallow(defaultView);
    expect(wrapper.childAt(0).prop('className')).toContain('label');
  });

  it('should have the class terraClinical-LabelValueView-value when given a text value', () => {
    const textInputView = (
      <LabelValueView label="Label" textValue="Sample Text" />
    );
    const wrapper = shallow(textInputView);
    expect(wrapper.childAt(1).prop('className')).toContain('value');
  });
});
