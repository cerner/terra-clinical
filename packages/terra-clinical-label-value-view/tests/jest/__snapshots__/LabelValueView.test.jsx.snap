exports[`LabelValueView should render LabelValueView for an node input 1`] = `
<div
  className="label-value-view">
  <div
    className="label">
    Label
  </div>
  <div>
    More Sample Text
  </div>
</div>
`;

exports[`LabelValueView should render LabelValueView for text and node inputs 1`] = `
<div
  className="label-value-view">
  <div
    className="label">
    Label
  </div>
  <div
    className="value">
    Sample Text 1
  </div>
  <div>
    More Sample Text
  </div>
</div>
`;

exports[`LabelValueView should render a LabelValueView for text input 1`] = `
<div
  className="label-value-view">
  <div
    className="label">
    Label
  </div>
  <div
    className="value">
    Sample Text
  </div>
</div>
`;

exports[`LabelValueView should render a default LabelValueView 1`] = `
<div
  className="label-value-view">
  <div
    className="label">
    Label
  </div>
  <div
    className="value">
    --
  </div>
</div>
`;
